src/main/java/net/minecraftforge/fluids/capability/wrappers/BlockWrapper.java

```diff
@@ -73,11 +71,11 @@ public int fill(FluidStack resource, FluidAction action)
                 BlockState state = world.getBlockState(blockPos);
                 if (liquidContainer.canPlaceLiquid(world, blockPos, state, resource.getFluid()))
                 {
-                    //If we are executing try to actually fill the container, if it failed return that we failed
-                    if (action.simulate() || liquidContainer.placeLiquid(world, blockPos, state, resource.getFluid().getFluidType().getStateForPlacement(world, blockPos, resource)))
+                    if (action.execute())
                     {
-                        return FluidType.BUCKET_VOLUME;
+                        liquidContainer.placeLiquid(world, blockPos, state, resource.getFluid().getFluidType().getStateForPlacement(world, blockPos, resource));
                     }
+                    return FluidType.BUCKET_VOLUME;
```

Comment:
Why did this need to be changed? I don't understand what this is doing differently other than saying that the value should always be the bucket volume when it asks whether it could place the liquid, which doesn't make sense if it can't actually be placed.

Reply:
The `liquidContainer` is an implementation of the vanilla interface `LiquidBlockContainer`, which is implemeted by waterloggable blocks. If such a block is waterlogged the `placeLiquid` method returns false, because no liquid is added to this block. But when you look into the `BucketItem` implementation, how `LiquidBlockContainer` is handled there, the result of the `placeLiquid` method is ignored:

```
         } else if (block instanceof LiquidBlockContainer && ((LiquidBlockContainer)block).canPlaceLiquid(p_150717_,p_150718_,blockstate,content)) {
            ((LiquidBlockContainer)block).placeLiquid(p_150717_, p_150718_, blockstate, ((FlowingFluid)this.content).getSource(false));
            this.playEmptySound(p_150716_, p_150717_, p_150718_);
            return true;
         } else {
```

Reply:
I think I understand the reasoning now because of a quirk of vanilla. I'll double check the solution and test when I awake in half a day since I currently don't have the headspace to properly review this. However, at a general glance, the solution seems fine.

Reply:
So, I don't have an issue with the solution. The main issue is that it is a breaking change in logic, regardless of expected behavior, so I'm not sure whether this should be merged into 1.19 or just held off until 1.20 in roughly 24-36 hours.

Reply:
Nice to hear that! :) I don't know, who can answer your question.^^ I can understand, that a breaking change shouldn't be merged into 1.19, but in my opinion it is a behaviour that should be there since vanilla 1.19.3. But how you decide is entirely up to you, :)